txpool/txpool.go

```diff
@@ -841,14 +842,19 @@ func (t *txPriceHeap) Length() uint64 {
 	return uint64(len(t.index))
 }
 
-func (t *txPriceHeap) Delete(tx *types.Transaction) {
+func (t *txPriceHeap) Delete(tx *types.Transaction) bool {
 	t.lock.Lock()
 	defer t.lock.Unlock()
 
-	if item, ok := t.index[tx.Hash]; ok {
-		heap.Remove(t.heap, item.index)
-		delete(t.index, tx.Hash)
+	item, ok := t.index[tx.Hash]
+	if !ok {
```

Comment:
you can simplify this additionally since you basically return value of the ok
if ok {
// do delete and heap.remove
}

return ok